index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+        device.
+    1. Set <code><a>this</a>.{{USBDevice/opened}}</code> to <code>false</code> and
+        <a>resolve</a> |promise|.
+
+    Note: When no [[!ECMAScript]] code can observe an instance of {{USBDevice}} |device|
+    anymore, the UA SHOULD run |device|.{{USBDevice/close()}}.
+</div>
+
+<div algorithm="USBDevice.selectConfiguration(configurationValue)">
+    The <dfn for=USBDevice method>selectConfiguration(|configurationValue|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following steps <a>in
+    parallel</a>:
+    1. If <a>this</a> is no longer connected to the system, <a>reject</a> |promise|
+        with a {{NotFoundError}} and abort these steps.
+    1. Let |selectedConfiguration| be <code>null</code>.
+    1. <a>For each</a> |configuration| of <a>this</a>.{{USBDevice/[[configurations]]}}:
+        1. If |configuration|.{{USBConfiguration/[[configurationValue]]}} is equal to
+            |configurationValue|, set |selectedConfiguration| to |configuration|
+            and break.
+    1. If |selectedConfiguration| is <code>null</code>, <a>reject</a> |promise|
+        with a {{NotFoundError}} and abort these steps.
+    1. If <code><a>this</a>.{{USBDevice/opened}}</code> is not equal to
+        <code>true</code> <a>reject</a> |promise| with an {{InvalidStateError}} and
+        abort these steps.
+    1. Let |activeConfiguration| be the result of <a>finding the current configuration</a>
+        with <a>this</a>.
+    1. If |activeConfiguration| is not <code>null</code>.
+        1. <a>For each</a> |interface| of |activeConfiguration|.{{USBConfiguration/[[interfaces]]}}:
+            1. <a>Abort transfers currently scheduled on an interface</a> with |interface|.
+    1. Issue a <code>SET_CONFIGURATION</code> <a>control transfer</a> with
+        <code>configurationValue</code> set to the |configurationValue|.
+        If this step fails <a>reject</a> |promise| with a {{NetworkError}} and
+        abort these steps.
+    1. Let |numInterfaces| be the <a>size</a> of |selectedConfiguration|.{{USBConfiguration/[[interfaces]]}}.
+    1. Resize <a>this</a>.{{USBDevice/[[selectedAlternateSetting]]}} to |numInterfaces|.
+    1. Fill <a>this</a>.{{USBDevice/[[selectedAlternateSetting]]}} with 0.
+    1. Resize <a>this</a>.{{USBDevice/[[claimedInterface]]}} to |numInterfaces|.
+    1. Fill <a>this</a>.{{USBDevice/[[claimedInterface]]}} with <code>false</code>.
+    1. Set <a>this</a>.{{USBDevice/[[configurationValue]]}} to |configurationValue| 
+        and <a>resolve</a> |promise|.
+</div>
+
+<div algorithm="USBDevice.claimInterface(interfaceNumber)">
+    The <dfn for=USBDevice method>claimInterface(|interfaceNumber|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following 
+    steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.